src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -103,11 +104,14 @@ private sealed class LatestDiagnosticsForSpanGetter
                 // updating the error list simultaneously.
                 var cacheFullDocumentDiagnostics = owner.AnalyzerService.GlobalOptions.IsPullDiagnostics(InternalDiagnosticsOptions.NormalDiagnosticMode);
 
+                // We log performance info when we are computing diagnostics for a span
+                // and also blocking for data, i.e. for lightbulb code path for "Ctrl + Dot" user command.
+                var logPerformanceInfo = range.HasValue && blockForData;
```

Comment:
📝 This is one of the locations where I'd want to see a gate on whether telemetry is enabled or not. We can avoid calculating the performance report during the blocking operation for users who disable telemetry.

Reply:
Instead of adding the check here, I added it to the `DiagnosticComputer`. This flag is passed as-is down to that type for all diagnostic computation paths, including lightbulb and background analysis, so we avoid any performance related calculation work for all diagnostic computation code paths.